vendor/psmodules/Cmder.ps1

```diff
@@ -1,18 +1,110 @@
-function Configure-Git($GIT_INSTALL_ROOT){
-  $env:Path += $(";" + $GIT_INSTALL_ROOT + "\cmd")
+function readVersion($gitPath) {
+    $gitExecutable = "${gitPath}\git.exe"
 
-  # Add "$GIT_INSTALL_ROOT\usr\bin" to the path if exists and not done already
-  $GIT_INSTALL_ROOT_ESC=$GIT_INSTALL_ROOT.replace('\','\\')
-  if ((test-path "$GIT_INSTALL_ROOT\usr\bin") -and -not ($env:path -match "$GIT_INSTALL_ROOT_ESC\\usr\\bin")) {
-      $env:path = "$env:path;$GIT_INSTALL_ROOT\usr\bin"
-  }
+    if (!(test-path "$gitExecutable")) {
+        return $null
+    }
+
+    $gitVersion = (cmd /c "${gitExecutable}" --version)
+
+    if ($gitVersion -match 'git version') {
+        ($trash1, $trash2, $gitVersion) = $gitVersion.split(' ', 3)
+    } else {
+        pause
+        return $null
+    }
+
+    return $gitVersion.toString()
+}
+
+function isGitShim($gitPath) {
+    # check if there's shim - and if yes follow the path
+
+    if (test-path "${gitPath}\git.shim") {
+      $shim = (get-content "${gitPath}\git.shim")
+      ($trash, $gitPath) = $shim.replace(' ','').split('=')
+
+      $gitPath=$gitPath.replace('\git.exe','')
+    }
+
+    return $gitPath.toString()
+}
+
+function compareVersions($userVersion, $vendorVersion) {
+    if (-not($userVersion -eq $null)) {
+        ($userMajor, $userMinor, $userPatch, $userBuild) = $userVersion.split('.', 4)
+    } else {
+        return -1
+    }
+
+    if (-not($vendorVersion -eq $null)) {
+        ($vendorMajor, $vendorMinor, $vendorPatch, $vendorBuild) = $vendorVersion.split('.', 4)
+    } else {
+        return 1
+    }
+
+    if ($userMajor -gt $vendorMajor) {return 1}
+    if ($userMajor -lt $vendorMajor) {return -1}
+
+    if ($userMinor -gt $vendorMinor) {return 1}
+    if ($userMinor -lt $vendorMinor) {return -1}
+
+    if ($userPatch -gt $vendorPatch) {return 1}
+    if ($userPatch -lt $vendorPatch) {return -1}
+
+    if ($userBuild -gt $vendorBuild) {return 1}
+    if ($userBuild -lt $vendorBuild) {return -1}
+
+    return 0
+}
+
+function compare_git_versions($userVersion, $vendorVersion) {
+    $result = compareVersions -userVersion $userVersion -vendorVersion $vendorVersion
+
+    # write-host "Compare Versions Result: ${result}"
+    if ($result -ge 0) {
+        return $userVersion
+    } else {
+        return $vendorVersion
+    }
+}
+
+function Configure-Git($gitRoot, $gitType, $gitPathUser){
```

Comment:
Prepending works if you are just prepending but this is truly reconfiguring the beginning and the middle of the path.

This does not just prepend the path where `git.exe` is but also the other two folders, `%GIT_INSTALL_ROOT%\usr\bin` and `%GIT_INSTALL_ROOT%\mingw64\bin`, the external Git install may or may not have added to the path either before or after the `c:\Windows;c:\Windows\system32` paths depending on what the user selected during install of Git for Windows.

I guess I could be less sneaky and replace `C:\Program Files\Git\usr\bin` with `%CMDER_ROOTvendor\git-for-windows\usr\bin;C:\Program Files\Git\usr\bin` and `%CMDER_ROOT%\vendor\git-for-windows\mingw64\bin;C:\Program Files\Git\mingw64\bin` leaving references to their original config in place instead of completely replacing it.

Reply:
Ah ok, you might find it more pleasant to work with an array instead of strings;
```powershell
# array from .split cannot have size modified. Cast into an arrayList,
[System.Collections.ArrayList]$modifiedPath = $env:Path.split(';')`

$modifiedPath.removeAt(<index>) #remove specific index or...
$modifiedPath.remove('<string to match>') # remove specific string from collection

$env:path = $modifiedPath -join ';'
```

Reply:
Did not look at this.  This was not easy to get to work and don't really have the time to completely rewrite it using arrays.

Reply:
If we don't want to merge because this is less than optimal I can see if I can carve out some time to look at it more closely.